### ISSUE TITLE
v3 is not working for boolean field

### ISSUE THREAD

**Issue by remy727** [okuramasafumi/alba#343]
## Describe the bug
`v3` is not working for the `boolean` field. But it works for `v2.4`

## To Reproduce
```rb
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # gem "rails"
  gem "activerecord", "~> 7.1.0"
  gem "sqlite3"
  gem "alba"
  gem "oj"
end


require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :subscriptions, force: true do |t|
    t.string(:name, null: false)
    t.boolean(:test, null: false, default: false)
  end
end

class Subscription < ActiveRecord::Base
  has_many :comments
end

# frozen_string_literal: true

class SubscriptionResource
  include Alba::Resource

  root_key :subscription, :subscriptions

  attributes :id,
    :name,
    :test
end

subscription = Subscription.create!(name: "Standard", test: false)
data = SubscriptionResource.new(subscription).serialize

```

## Expected behavior
It works without any issue.

## Actual behavior
```
no implicit conversion of Subscription into Integer (TypeError)
```
![image](https://github.com/okuramasafumi/alba/assets/69775411/c306f4d5-c54a-42e1-a0e8-17d62501e8d1)

## Environment
- Ruby version: `3.2.1`
- Rails version: `7.1.1`

**Comment by okuramasafumi** [comment]
@remy727 When I changed `test` to `test1`, the problem seemed resolved. I believe it has the same root cause as #342.

https://docs.ruby-lang.org/en/3.2/Kernel.html#method-i-test
I haven't used it, but there's `Kernel#test` method and it seems this method was called instead of `test` attribute.

**Comment by remy727** [comment]
Aha, got it. Thanks for taking a look on it.

**Comment by okuramasafumi** [comment]
Fixed by #344
